python/examples/token_swap/token_swap.py

```diff
@@ -0,0 +1,245 @@
+from xrpl_plugin.transactors import (
+    ConsequencesFactoryType,
+    preflight2,
+)
+
+from xrpl_plugin.models import (
+    Transactor,
+    LedgerObject,
+    Amendment,
+    TERCode,
+)
+from xrpl_plugin.basic_types import (
+    soeREQUIRED,
+    VoteBehavior,
+)
+from xrpl_plugin.keylets import (
+    account_keylet,
+    trustline_keylet,
+    Keylet,
+    owner_dir_keylet,
+)
+from xrpl_plugin.return_codes import (
+    tecINSUFFICIENT_RESERVE,
+    tecINTERNAL,
+    temBAD_AMOUNT,
+    tesSUCCESS,
+    tecDIR_FULL,
+)
+
+from xrpl_plugin.sfields import (
+    sf_balance,
+    create_new_sfield,
+    sf_account,
+    sf_amount,
+    sf_owner_count,
+    sf_owner_node,
+)
+from xrpl_plugin.stypes import (
+    STAmount,
+    STAccount,
+    STUInt64,
+    index_hash,
+)
+
+from xrpl_plugin.ledger_objects import (
+    adjust_owner_count,
+    make_sle,
+)
+
+sf_amount_other = create_new_sfield(STAmount, "AmountOther", 33)
+sf_account_other = create_new_sfield(STAccount, "AccountOther", 35)
+sf_token_swap_id = create_new_sfield(STUInt64, "TokenSwapId", 36)
+
+sfields = [sf_amount_other, sf_account_other, sf_token_swap_id]
+
+ltTOKEN_SWAP_PROPOSE = 0x77
+NEW_TOKEN_SWAP_NAMESPACE = ord("w")
+
+temINVALID_TOKEN_SWAP_ID = -256
+ter_codes = [TERCode(temINVALID_TOKEN_SWAP_ID, "temINVALID_TOKEN_SWAP_ID", "The token swap is incorrect or has ben accepted")]
+
+def new_token_swap_keylet(src, seq):
+    return Keylet(ltTOKEN_SWAP_PROPOSE, index_hash(NEW_TOKEN_SWAP_NAMESPACE, src, seq))
+
+def visit_entry_xrp_change_token_swap(is_delete, entry, is_before):
+    if is_before:
+        return -1 * entry[sf_amount].xrp().drops
+    if is_delete:
+        return 0
+    return entry[sf_amount].xrp().drops
+
+ledger_objects = [
+    LedgerObject(
+        object_type=ltTOKEN_SWAP_PROPOSE,
+        name="TokenSwap",
+        rpc_name="token_swap",
+        object_format=[
+            (sf_account, soeREQUIRED),
+            (sf_amount, soeREQUIRED),
+            (sf_token_swap_id, soeREQUIRED),
+            (sf_amount_other, soeREQUIRED),
+            (sf_account_other, soeREQUIRED),
+            (sf_owner_node, soeREQUIRED),
+        ],
+        is_deletion_blocker=True,
+        # visit_entry_xrp_change=visit_entry_xrp_change_token_swap,
```

Comment:
Is there a reason this is commented out?

Reply:
no, my bad